### ISSUE TITLE
Error due to Floating Point inaccuracies

### ISSUE THREAD

**Issue by shivin9** [shivin9/PDEOperators.jl#3]
The sum of calculated stencil doesn't come out to be exactly 0 with the error increasing as we increase the order of approximation.
For a stencil for 4th order derivative and 10th order approximation, we have
```julia
julia> sum(A.stencil_coefs)
-1.4052778038453617e-14
```
This is not a problem with Float64 as tested with BigFloat also.
So the solution is to adjust the middle coefficient of the stencil to set the overall sum = 0.
Reference:
https://scicomp.stackexchange.com/questions/11249/numerical-derivative-and-finite-difference-coefficients-any-update-of-the-fornb

```julia
julia> A.stencil_coefs[7] -= sum(A.stencil_coefs)
1.371740740740740209188154585806529439651058055460453033447265625000000000000000e+01

julia> sum(A.stencil_coefs)
0.000000000000000000000000000000000000000000000000000000000000000000000000000000

julia> res[500]
2.399999481622449663473228831378492031944915652275085449218750000000000000000000e+01
```
which is what we want.

**Comment by dextorious** [comment]
Good catch. In general, I've never seen a practical case (like a PDE discretization) where this numerical issue would matter (say, by significantly impacting the numerical stability), but the fix is cheap and there's no reason not to do it.

**Comment by shivin9** [comment]
[Reminder to self]: Use @code_warntype
